set-client/src/main/java/eu/europeana/set/client/config/ClientConfiguration.java

```diff
@@ -1,126 +1,76 @@
 package eu.europeana.set.client.config;
 
 import java.io.IOException;
-import java.io.InputStream;
 import java.util.Properties;
 
-import eu.europeana.set.client.exception.TechnicalRuntimeException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 /**
  * configuration for accessing remote api
- * 
  * @author GordeaS
- *
  */
 
 public final class ClientConfiguration {
 
+    private static final Logger LOGGER = LogManager.getLogger(ClientConfiguration.class);
+
     protected static final String SET_CLIENT_PROPERTIES_FILE = "/set-client.user.properties";
-    protected static final String PROP_SET_API_KEY = "set.api.key";
-    protected static final String PROP_SET_SERVICE_URI = "set.service.uri";
-    protected static final String PROP_OAUTH_SERVICE_URI = "oauth.service.uri";
-    protected static final String PROP_OAUTH_REQUEST_PARAMS = "oauth.token.request.params";
+    public static final String PROP_SET_API_KEY = "set.api.key";
+    public static final String PROP_SET_SERVICE_URI = "set.service.uri";
+    public static final String PROP_OAUTH_SERVICE_URI = "oauth.service.uri";
+    public static final String PROP_OAUTH_REQUEST_PARAMS = "oauth.token.request.params";
 
     private Properties properties;
-    private static ClientConfiguration singleton;
-
-    /**
-     * Hide the default constructor
-     */
-    private ClientConfiguration() {
-    }
 
     /**
-     * Accessor method for the singleton
-     * 
-     * @return
+     * Creates ClientConfiguration instance with set client properties
      */
-    public static synchronized ClientConfiguration getInstance() {
-	if (singleton == null) {
-	    singleton = new ClientConfiguration();
-	    singleton.loadProperties();
-	}
-
-	return singleton;
+    public ClientConfiguration() {
+        loadProperties(SET_CLIENT_PROPERTIES_FILE);
```

Comment:
is there a reason to change the configuration to not be a singleton anymore? We should avoid reading the properties files multiple times as that can generate major performance loss

Reply:
Usually, we use spring boot and create a bean, only a single instance is always created. 
Also, when creating a client for any API, we create an instance of the Client class, and that class should be able to accept a configuration class. 
` public UserSetApiClient(ClientConfiguration configuration)`

Reply:
yes, but the properties still need to be read and currently we only support properties from the classpath